### PR TITLE
Run command validation on all commands.

### PR DESCRIPTION
**Pull request by tvolkert**
This makes command validation happen as part of `verifyThenRunCommand()`,
using a newly introduced protected method (`validateCommand()`) rather than
a `commandValidator` property (that subclasses were responsible for manually
invoking).

### CODE REVIEW

File: packages/flutter_tools/test/commands/drive_test.dart [hunk 1]
```diff
@@ -76,10 +76,11 @@ void main() {
 
       final String testApp = fs.path.join(cwd.path, 'test', 'e2e.dart');
       final String testFile = fs.path.join(cwd.path, 'test_driver', 'e2e_test.dart');
+      fs.file(testApp).createSync(recursive: true);
 
       final List<String> args = <String>[
         'drive',
-        '--target=$testApp}',
```
Comment (cbracken) [inline packages/flutter_tools/test/commands/drive_test.dart]: Nice catch.

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm

**Comment by cbracken** [comment]
Any new tests on build/install/test?

**Comment by tvolkert** [comment]
> Any new tests on build/install/test?

No need - all I did was move code around, not really change anything

**Comment by Hixie** [comment]
@abarth reports that this broke Fuchsia. We should probably revert.

**Comment by tvolkert** [comment]
> @abarth reports that this broke Fuchsia. We should probably revert.

Negative - @abarth and I just identified #12237 as the culprit

**Comment by abarth** [comment]
This clarify: this change did break Fuchsia, but I was able to fix it.  #12237 also broke Fuchsia and I was not able to fix it easily.  That change doesn't need to be reverted either because Fuchsia now pins the revision of flutter.git that it uses.

**Pull request merged**